packages/workflow/src/Workflow.ts

```diff
@@ -1421,6 +1424,74 @@ export class Workflow {
 			};
 		}
 	}
+
+	/**
+	 * Select the node to use as starter for a subworkflow execution.
+	 */
+	static selectSubworkflowStarter(nodes: INode[]) {
+		const found = nodes.find((n) => SUBWORKFLOW_STARTER_NODES.includes(n.type));
```

Comment:
This function used to prioritize for the `executeWorkflowTrigger` node and now we'll fetch any initial node. This might break existing behavior.

Reply:
`Array.prototype.find` goes through the array in order and `executeWorkflowTrigger` is the zeroth element in `SUBWORKFLOW_STARTER_NODES` so the result is the same.

Reply:
Indeed but let's say we have the follwing:

```
[
    'set node',
    'webhook trigger node',
    'execute workflow trigger node',
]
```

This would iterate and return the `webhook trigger node` instead of the desired `execute workflow trigger node`, right?

Reply:
Thank you, please see [`94780d2`](https://github.com/n8n-io/n8n/pull/8305/commits/94780d27c18415885b9e22b50ed9b87d448d7bc2)